framework/include/base/MooseObject.h

```diff
@@ -82,6 +152,9 @@ class MooseObject :
   /// The name of this object, reference to value stored in InputParameters
   const std::string & _name;
 
+  /// Reference to the "enable" InputParaemters, used by Controls for toggling on/off MooseObjects
+  const bool & _enabled;
```

Comment:
While I like this idea. It does enable a whole new class of bugs. If objects can be turned on and off at will. The integrity of the system cannot be assured.